core/src/main/java/org/jdbi/v3/core/argument/EnumByNameArgumentFactory.java

```diff
@@ -14,21 +14,25 @@
 package org.jdbi.v3.core.argument;
 
 import java.lang.reflect.Type;
+import java.sql.Types;
 import java.util.Optional;
-
-import org.jdbi.v3.core.argument.internal.strategies.LoggableBinderArgument;
+import org.jdbi.v3.core.EnumByName;
 import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.internal.Enums;
 
-class EnumArgumentFactory implements ArgumentFactory {
+@EnumByName
+class EnumByNameArgumentFactory implements ArgumentFactory {
```

Comment:
I think it would be much simpler to change this class (which is package-private, so this would not break compatibility) to implement `QualifiedArgumentFactory` instead of `ArgumentFactory`.

The precedence rules could live in e.g. `EnumConfig.resolveEnumStrategy(QualifiedType)`, which would return the correct `EnumStrategy` enum value based on the precedence rules:

```java
public EnumStrategy resolveEnumStrategy(QualifiedType type) {
  EnumStrategy strategy = resolveEnumStrategy(type.getQualifiers());

  if (strategy == null) {
    Class<?> enumType = getErasedType(type.getType());
    strategy = resolveEnumStrategy(getQualifiers(enumType));
  }

  if (strategy == null) {
    strategy = defaultStrategy;
  }
  
  return strategy;
}

private EnumStrategy resolveEnumStrategy(Collection<Annotation> qualifiers) {
  if (qualifiers.contains(EnumByName.class)) {
    return EnumStrategy.BY_NAME;
  }

  if (qualifiers.contains(EnumByOrdinal.class)) {
    return EnumStrategy.BY_ORDINAL;
  }

  return null;
}
```

Then the argument factory and mapper factory can delegate to that config class to resolve the enum strategy, and just pick the existing argument / mapper implementation according to the resolved strategy.